packages/slice-machine/server/src/api/screenshots/generate.ts

```diff
@@ -29,62 +23,30 @@ export async function generateScreenshotAndRemoveCustom(
   sliceName: string,
   variationId: string
 ): Promise<ScreenshotResults> {
-  const { screenshots, failure } = await generateScreenshot(
-    env,
-    libraryName,
-    sliceName,
-    variationId
-  );
-
-  removeCustomScreenshot(env, libraryName, sliceName, variationId);
-
-  return {
-    screenshots: screenshots,
-    failure: failure,
-  };
-}
-
-export async function generateScreenshot(
-  env: BackendEnvironment,
-  libraryName: string,
-  sliceName: string,
-  variationId: string
-): Promise<ScreenshotResults> {
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   const slice = IO.Slice.readSlice(
     NodeUtils.CustomPaths(env.cwd).library(libraryName).slice(sliceName).model()
   );
 
-  const promises: Promise<ScreenshotUI>[] = [variationId].map(
-    (id: VariationSM["id"]) => generateForVariation(env, libraryName, slice, id)
-  );
-
-  const results = await Promise.allSettled(promises);
-
-  return results.reduce(
-    (acc: ScreenshotResults, result: PromiseSettledResult<ScreenshotUI>) => {
-      const key = variationId;
-
-      const screenshots =
-        result.status === "fulfilled"
-          ? {
-              ...acc.screenshots,
-              [key]: result.value,
-            }
-          : acc.screenshots;
-
-      const failure =
-        result.status === "rejected"
-          ? [
-              ...acc.failure,
-              { error: result.reason as Error, variationId: key },
-            ]
-          : acc.failure;
-
-      return { screenshots, failure };
-    },
-    { screenshots: {}, failure: [] } as ScreenshotResults
-  );
+  try {
+    const result = await generateForVariation(
+      env,
+      libraryName,
+      slice,
+      variationId
+    );
+
+    removeCustomScreenshot(env, libraryName, sliceName, variationId);
+
+    return {
+      screenshots: [result],
```

Comment:
almost perfect, just if we know there is going to be only one screenshot here I would return a `ScreenshotUI | null`, that way in the frontend using the variationID you already have, the state can be updated easily.